libmscore/chordlist.cpp

```diff
@@ -1464,6 +1464,122 @@ ChordDescription::ChordDescription(const QString& name)
       exportOk = false;
       }
 
+//---------------------------------------------------------
+//   intervals
+//    returns a QList of integers holding the intervals
+//    from the root note
+//---------------------------------------------------------
+QList<int> ChordDescription::intervals() const {
+      //TODO - PHV take a look again later, this might be uneeded
+      QList<int> ret;
+      int k = chord.getKeys();
+      int n = 1;
+      k >>= 1;
+      while (k > 0) {
+            if (k & 1)
+                  ret << n;
+            n++;
+            k >>= 1;
+            }
+      return ret;
+      }
+
+//---------------------------------------------------------
+//   noteTpcs
+//    returns a QList of note tpcs in the chord ordered by
+//    pitch
+//---------------------------------------------------------
+QList<int> ChordDescription::noteTpcs(int rootTpc) const {
+      //TODO - PHV take a look again later, this isn't super efficient
```

Comment:
I would also say, I don't think the way you are going about this is generally right.  Instead of trying guess at a good spelling for a note based on what other notes are in the chord, you should be looking directly at the parsed representation - that's (part of) what it's there for.  Eg, for the interval 3, you call it a minor third iff and only if the chord is a minor chord according to the parsed representation.  Maybe there are some corner cases where looking at another pitch might help differentiate something, but that should be only a fallback if the parsed representation isn't sufficient for whatever reason.

Reply:
Also, unless I'm missing something, I don't see anything to differentiate things like #5 vs b13 or #11 vs b5 - all of this information is right there in the parsed chord, though.

Reply:
Ah okay so would you suggest still using the HChord to use the quality, extension, and alterations to build up the TPCs and intervals from strings?

Reply:
I'm not sure, to be honest.  I'd say if you were *only* doing the behind-the-scenes playback, the HChord would make sense, but since you really need to generate things yourself anyhow for display purposes, you're better off using the parsed chord that as your single source of info, then we know everything is consistent.